docs/source/task-launch.rst

```diff
@@ -84,71 +105,131 @@ We can submit tasks from other tasks.  This allows us to make decisions while
+
+``dask.delayed``
+~~~~~~~~~~~~~~~~
+
+``dask.delayed`` behaves as normal: it submits the functions to the graph,
+optimizes for less bandwidth/computation and gathers the results.
+For more detail, see `dask.delayed`_.
 
 .. code-block:: python
 
-   from distributed import worker_client
+    In [1]: from distributed import Client
+    In [2]: from dask import delayed, compute
 
-   def process_all(data):
-       with worker_client() as e:
-           elements = e.scatter(data)
-           futures = e.map(process_element, elements)
-           analysis = e.submit(aggregate, futures)
-           result = analysis.result()
-       return result
+    In [3]: # these features require the newer dask.distributed scheduler
+    In [4]: client = Client()
 
-    analysis = e.submit(process_all, data)  # spawns many tasks
+    In [4]: @delayed
+       ...: def fib(n):
+       ...:     if n < 2:
+       ...:         return n
+       ...:     # We can use dask.delayed and dask.compute to launch
+       ...:     # computation from within tasks
+       ...:     a = fib(n-1)  # these calls are delayed
+       ...:     b = fib(n-2)
+       ...:     a, b = compute(a, b)  # execute both in parallel
+       ...:     return a + b
 
-This approach is somewhat complex but very powerful.  It allows you to spawn
-tasks that themselves act as potentially long-running clients, managing their
-own independent workloads.
+    In [5]: fib(10).compute()
+    Out[5]: 55
 
-Extended Example
-~~~~~~~~~~~~~~~~
+.. _dask.delayed: https://dask.pydata.org/en/latest/delayed.html
 
-This example computing the Fibonacci numbers creates tasks that submit tasks
-that submit tasks that submit other tasks, etc..
+``get_client``
+~~~~~~~~~~~~~~
+
+``distributed.get_client`` allows individual workers to access the ``client``
+that the scheduler uses:
 
 .. code-block:: python
 
-   In [1]: from distributed import Client, worker_client
+    In [1]: from distributed import Client, get_client, secede, rejoin
 
-   In [2]: client = Client()
+    In [2]: def fib(n):
+       ...:     if n < 2:
+       ...:         return n
+       ...:     client = get_client()
+       ...:     a_future = client.submit(fib, n - 1)
+       ...:     b_future = client.submit(fib, n - 2)
+       ...:     a, b = client.gather([a_future, b_future])
+       ...:     return a + b
 
-   In [3]: def fib(n):
-      ...:     if n < 2:
-      ...:         return n
-      ...:     else:
-      ...:         with worker_client() as c
-      ...:             a = c.submit(fib, n - 1)
-      ...:             b = c.submit(fib, n - 2)
-      ...:             a, b = c.gather([a, b])
-      ...:             return a + b
-      ...:
+    In [3]: client = Client()
+
+    In [4]: future = client.submit(fib, 10)
+
+    In [5]: future.result()
+    Out[5]: 55
+
+This can deadlock the scheduler if too many nodes request jobs at once. To
```

Comment:
I would add in a `"However,"` to start here.  We might also explain a bit more about the mechanism that causes deadlocking.  It's not just requesting jobs that's the problem, it's that they tasks doing the requesting, like our call to `fib(55)` are taking up a slot in a worker and if enough tasks like that occur then all tasks will just be waiting for slots that will never appear.